tests/ut_magma_main.cc

```diff
@@ -0,0 +1,54 @@
+#define CATCH_CONFIG_RUNNER
+// Disable colored output because it doesn't play well with ctest
+#define CATCH_CONFIG_COLOUR_NONE
+#include "catch.hpp"
+
+#include <mpi.h>
+
+#ifdef HAVE_MAGMA
```

Comment:
Since you have all there `ifdef` why not modifying `ut_main.cc` instead.

Reply:
if you modify ut_main.cc, then all the tests based on that file need to link with magma. I was trying to avoid that